### ISSUE TITLE
Handle missing dependencies/dev_dependencies entry in add command

### ISSUE THREAD

**Issue by shyndman** [shyndman/pubx#7]
It's valid to not have `dependencies` or `dev_dependencies` in a pubspec, but right now we're assuming that they're going to be there. The `add` command should add them if it needs them.

**Comment by Vanethos** [comment]
Should this always be added at the end of the file?

**Comment by shyndman** [comment]
That seems reasonable.

**Comment by rhalff** [comment]
I've just experienced this issue:

Without a `dependencies:` in `pubspec.yaml`:
```bash
pubx add meta
There was a problem adding the dependency to your project
An unknown error has occured.
The missing pub commands.

Usage: pubx <command> [arguments]

Global options:
-h, --help    Print this usage information.

Available commands:
  add      Add a package to your pubspec.yaml file
  help     Display help information for pubx.
  search   Searches pub.dev for packages
  view     View information about a pub.dev package
  which    Locate a dependency

Run "pubx help <command>" for more information about a command.
Unhandled exception:
NoSuchMethodError: The method 'containsKey' was called on null.
Receiver: null
Tried calling: containsKey("meta")
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:50:5)
#1      AddCommand.run (package:pubx/src/commands/add.dart:61:27)
<asynchronous suspension>
#2      CommandRunner.runCommand (package:args/command_runner.dart:197:27)
<asynchronous suspension>
#3      CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
#4      new Future.sync (dart:async/future.dart:224:31)
#5      CommandRunner.run (package:args/command_runner.dart:112:14)
#6      main (file:///home/rhalff/.pub-cache/hosted/pub.dartlang.org/pubx-1.2.0/bin/pubx.dart:15:18)
<asynchronous suspension>
#7      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#8      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)
```

With an empty `dependencies:` in `pubspec.yaml`:
```bash
pubx add meta    
There was a problem adding the dependency to your project
An unknown error has occured.
The missing pub commands.

Usage: pubx <command> [arguments]

Global options:
-h, --help    Print this usage information.

Available commands:
  add      Add a package to your pubspec.yaml file
  help     Display help information for pubx.
  search   Searches pub.dev for packages
  view     View information about a pub.dev package
  which    Locate a dependency

Run "pubx help <command>" for more information about a command.
Unhandled exception:
type 'YamlScalar' is not a subtype of type 'YamlMap' in type cast
#0      AddCommand.run (package:pubx/src/commands/add.dart:55:62)
<asynchronous suspension>
#1      CommandRunner.runCommand (package:args/command_runner.dart:197:27)
<asynchronous suspension>
#2      CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:112:25)
#3      new Future.sync (dart:async/future.dart:224:31)
#4      CommandRunner.run (package:args/command_runner.dart:112:14)
#5      main (file:///home/rhalff/.pub-cache/hosted/pub.dartlang.org/pubx-1.2.0/bin/pubx.dart:15:18)
<asynchronous suspension>
#6      _startIsolate.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:299:32)
#7      _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:12)

```

**Comment by shyndman** [comment]
Sorry about that @rhalff. That happened sooner than I'd hoped. :)

Let me take a look.

**Comment by shyndman** [comment]
Published as 1.2.1

**Comment by rhalff** [comment]
Thanks!
